Vizzuality/marxan-cloud
# feat(front): view mode based on permissions [MARXAN-1295]

## View mode based on permissions

### Overview

This PR contains:
- [x] New hooks folder for permissions with calls to set user access to the view depending on their role (viewer contributor and owner) and if the scenario is locked.
- [x] A view has been implemented for cases where the user cannot edit content (when the stage is locked, when the user is a viewer or in some cases where the user is a contributor).

### Designs

_At the moment we don't have designs for the viewer mode, it's made without being based on designs._

### Testing instructions

To test, please consult [Role Based Access Control table](https://github.com/Vizzuality/marxan-cloud/blob/develop/docs/features/role-based-access-control/high-level-design.md) and check `project_viewer`, `project_contributor` and `project_owner roles`. 

Keep in mind that the rest of the roles will be implemented in the future.

In addition, to test the functionality, it is essential to open two different user profiles to be able to verify that the user who is editing can access all the functionalities, and the user who is trying to access the same scenario and is locked, cannot edit.

### Feature relevant tickets

[MARXAN-1295](https://vizzuality.atlassian.net/browse/MARXAN-1295)
